R/g_lineplot.R

```diff
@@ -0,0 +1,246 @@
+#'
+#' @import ggplot2
+#' @import dplyr
+#' @export
+g_lineplot <- function(data,
+                       x_var,
+                       y_var,
+                       group_var,
+                       colorby_var,
+                       facet_by = NULL,
+                       ylog_scale = FALSE,
+                       show_threshold = FALSE,
+                       threshold_value = 0,
+                       show_dose = FALSE,
+                       dose_data = NULL,
+                       palette = NULL,
+                       show_sd_min = FALSE,
+                       show_sd_max = FALSE,
+                       show_ci = FALSE,
+                       tooltip_vars = NULL,
+                       labels_df = NULL) {
+
+  # Set up plot labels - assumes individual plot unless mean columns are present
+  is_mean_plot <- all(c("Mean", "SD", "N") %in% names(data))
+
+  # Defensively set mean-plot flags to FALSE if it's not a mean plot
+  if (!is_mean_plot) {
+    show_sd_min <- FALSE
+    show_sd_max <- FALSE
+    show_ci     <- FALSE
+  }
+
+  if (is_mean_plot) {
+    x_lab <- paste0("Nominal Time [", unique(data$RRLTU), "]")
+    y_lab <- paste0("Mean Concentration [", unique(data$AVALU), "]")
+    title <- "Mean PK Concentration - Time Profile"
+  } else {
+    x_lab <- paste0("Time [", unique(data$RRLTU), "]")
+    y_lab <- paste0("Concentration [", unique(data$AVALU), "]")
+    title <- "PK Concentration - Time Profile"
+  }
+
+  # --- Tooltip Construction ---
+  if (nrow(data) == 0) {
+    data$tooltip_text <- character(0)
+  }
+
+  if (!is.null(tooltip_vars)) {
+    if (!is.null(labels_df)) {
+      # Generate tooltip if labels_df available
+      data$tooltip_text <- generate_tooltip_text(data, labels_df, tooltip_vars, "ADNCA")
+    } else {
+      # Fallback to simple paste if labels_df is missing
+      valid_vars <- intersect(tooltip_vars, names(data))
+      if (length(valid_vars) > 0) {
+        parts <- lapply(valid_vars, function(v) paste0(v, ": ", data[[v]]))
+        data$tooltip_text <- do.call(paste, c(parts, sep = "<br>"))
+      }
+    }
+  } else {
+    data$tooltip_text <- rep(NA_character_, nrow(data))
+  }
+
+  plt <- ggplot(data, aes(
+    x = .data[[x_var]],
+    y = .data[[y_var]],
+    color = color_var,
+    group = .data[[group_var]],
+    text = tooltip_text
+  )) +
+    geom_line() +
+    geom_point() +
+    labs(
+      x = x_lab,
+      y = y_lab,
+      title = title,
+      color = paste(colorby_var, collapse = ", ")
+    ) +
+    theme_bw()
+
+  # Add optional layers
+  optional_layers <- list(
+    .add_palette(palette),
+    .add_y_scale(ylog_scale),
+    .add_faceting(facet_by),
+    .add_threshold(show_threshold, threshold_value),
+    .add_dose_lines(show_dose, dose_data, facet_by),
+    .add_mean_layers(
+      is_mean_plot,
+      show_sd_min,
+      show_sd_max,
+      show_ci,
+      colorby_var,
+      y_var,
+      x_var,
+      group_var
+    )
+  )
+
+  return(plt + optional_layers)
```

Comment:
style: implicit return
```suggestion
  plt + optional_layers
```